app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt

```diff
@@ -224,18 +227,37 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
   // Advice user to turn on hotspot manually for API<26
   private fun startHotspotManuallyDialog() {
 
-    alertDialogShower.show(KiwixDialog.StartHotspotManually,
-      ::launchTetheringSettingsScreen,
-      { startActivity(Intent(Settings.ACTION_WIFI_SETTINGS)) },
-      {
-        progressDialog = ProgressDialog.show(
-          this,
-          getString(R.string.progress_dialog_starting_server), "",
-          true
-        )
-        startService(createHotspotIntent(ACTION_CHECK_IP_ADDRESS))
-      }
+    val wifiManager =
```

Comment:
This should be provided by dagger and we may want to even wrap it in another class

Reply:
@macgills could you please give some hints about the part of wrapping WifiManger in another class?

Reply:
if you had a class like so

```
class WrapperClass @Inject constructor(val realObject: RealObject){

fun doSomething(){
//lots of method calls to realObject
}
```

If Dagger knows how to create a `RealObject` then in

```

class AnUnrelatedClass{

@Inject lateinit var WrapperClass wrapperclass

init{
getTheRightDaggerComponent().inject(this)
}
}
```
AnUnrelatedClass will get/create an instance of `WrapperClass` and put it into `wrapperClass` for you because you called inject. This is field Injection.

DI can be a pretty intimidating topic but the tutorial [here](https://developer.android.com/training/dependency-injection
) is actually alright

Reply:
@macgills I tried to do this. But I'm getting the following error right now:

```
/home/yashk2000/Projects/kiwix-android/app/build/tmp/kapt3/stubs/debug/org/kiwix/kiwixmobile/di/components/KiwixComponent.java:8: error: [Dagger/MissingBinding] android.net.wifi.WifiManager cannot be provided without an @Inject constructor or an @Provides-annotated method.
public abstract interface KiwixComponent {
                ^
      android.net.wifi.WifiManager is injected at
          org.kiwix.kiwixmobile.core.utils.WifiManagerUtils(wifiManager)
      org.kiwix.kiwixmobile.core.utils.WifiManagerUtils is injected at
          org.kiwix.kiwixmobile.webserver.ZimHostActivity.wifiManagerUtils
      org.kiwix.kiwixmobile.webserver.ZimHostActivity is injected at
```

Could you please help out?

Reply:
You need to tell dagger how to create a WifiManager

>[Dagger/MissingBinding] android.net.wifi.WifiManager cannot be provided without an @Inject constructor or an @Provides-annotated method.

You can't use `@inject` on a constructor because it is not a class we own

Look for other system services in our dagger modules, they should provide an example

Reply:
@macgills is this fine now?

Reply:
Yup, DI all looks good

Reply:
@macgills thanks for the help with the DI :)